components/Accordion/vue/Accordion.vue

```diff
@@ -4,8 +4,14 @@
       class="flex items-center"
       :class="[CssClasses.summary, headingClassName ?? CssClasses.summaryColor]"
     >
-      <Icon v-if="props.icon" :class="CssClasses.icon" />
-      <hr v-if="props.icon && separator" :class="CssClasses.separator" />
+      <span v-if="slots.iconEl" :class="CssClasses.icon">
+        <slot name="iconEl" />
+      </span>
+      <Icon v-else-if="props.icon" :class="CssClasses.icon" />
+      <hr
+        v-if="(props.icon || slots.iconEl) && separator"
+        :class="CssClasses.separator"
+      />
```

Comment:
Use the default slot value instead of complicated js work

```suggestion
      <slot name="iconEl">
        <Icon :class="CssClasses.icon" />
      </slot>
      <hr
        v-if="(props.icon || slots.iconEl) && separator"
        :class="CssClasses.separator"
      />
```